### DISCUSSION TITLE
Serial port support?

### DISCUSSION THREAD

**Discussion by NilsKr** [OpenCBM/OpenCBM#114] [Ideas]
Hi there,
Just found this project. Are there any plans to support COM serial port communication instead of USB using a dedicated driver?
I'm thinking 64 bits windows support here.
Thank you and sorry if this topic has been discussed before - I haven't found anything about this yet.
Best regards,
Nils

**Comment by spiro-trikaliotis** [comment]
What would be the benefit of this? Only to avoid the custom driver?

**Reply by NilsKr** [reply]
Thank you for asking. Yes, that was the idea. With 32 bits Windows practically only available in isolated virtual machine because of security issues, I'd think that having an option that doesn't need a custom USB driver would be useful. Or a generic request/response API so one could, for instance, redirect the communication to an HTTP connection (e.g. an ESP32 based interface with Wifi - this would be beneficial to electrically de-couple the disk drive from the host computer).
Note: I have hardly had any time so far to get to understand the architecture of your project, so it could very well be that I have a wrong/incomplete mental image. In that case sorry to bother you.

**Reply by spiro-trikaliotis** [reply]
The 64 bit driver will install on Windows 10 without problems, as far as I know, if you use the install script install.cmd and let it install the driver.

Don't you agree?

So, why do you want to use 32 bit drivers?

**Reply by NilsKr** [reply]
I read somewhere, but I can't find the source anymore, that there will be issues on Windows 11 due to driver signing. Since Win10 is basically end of life, I assume that there may be difficult times ahead. But maybe (hopefully!) I read it wrongly. I for sure don't want to use 32 bits drivers, but it seemed that this might be a requirement in the near future.
